lib/reporters/dot.js

```diff
@@ -19,9 +19,13 @@ exports = module.exports = Dot;
  *
  * @api public
  * @param {Runner} runner
+ * @param {Object} options
+ * @param {any} options.reporterOptions.showErrorsImmediately - if truthy, errors will be shown
+ * immediately after tests fail instead of after all tests have finished.
  */
-function Dot (runner) {
+function Dot (runner, options) {
```

Comment:
You know what?  We don't actually use anything in `options` passed to these reporters *except* `reporterOptions`, so I think we could just refactor across the board so it's not nested this way within another object.  You're welcome to refactor it, or not.  :wink:

Reply:
@boneskull well, even if you guys don't use it, some 3rd-party reporters do for better or worse, so that would have to be `semver-major`.

Reply:
speaking of 3rd-party reporters--what's stopping you from taking this change and making it into a 3rd-party reporter?